src/sql-parser/src/parser.rs

```diff
@@ -2532,7 +2532,21 @@ impl Parser {
 
     fn parse_show(&mut self) -> Result<Statement, ParserError> {
         if self.parse_keyword("DATABASES") {
+            let select = Select::default()
+                .from(TableWithJoins {
+                    relation: TableFactor::Table {
+                        name: ObjectName(vec![Ident::new("mz_databases")]),
+                        alias: None,
+                    },
+                    joins: vec![],
+                })
+                .project(SelectItem::Expr {
+                    expr: Expr::Identifier(vec![Ident::new("database")]),
+                    alias: None,
+                });
```

Comment:
This snippet looks great, but putting it in the parser is a bit weird! It couples the SQL parser to the details of the `mz_catalog` views, which isn't a coupling that exists before this commit. But I think it should be super straightforward to just leave the AST as-is but move this bit to the `handle_show_databases` function in the planner?